feat: classify validation issues by category (schema vs. NTIA compliance)

## Summary

When running `sbom-validator validate`, all findings are currently surfaced as a flat list of `ValidationIssue` entries distinguished only by severity (`ERROR`, `WARNING`, `INFO`). There is no structural or visual distinction between:

- **Schema validation errors** — the file does not conform to the declared format schema (e.g. CycloneDX JSON schema, SPDX JSON schema)
- **NTIA compliance issues** — the file passes schema validation but is missing one or more of the NTIA minimum elements (supplier, component name, version, unique identifier, dependency relationships, author, timestamp)

## Motivation

During positive and negative testing of the validation pipeline, the lack of categorisation made it harder to quickly understand *why* a file failed:

- A schema error means the file is structurally malformed — tooling that produced it is broken.
- An NTIA issue means the file is structurally valid but incomplete for supply-chain use cases — a policy/content gap, not a format bug.

Treating them identically forces the operator to read every message and mentally categorise them. Adding an explicit `category` (or `issue_type`) field to `ValidationIssue` would let both human readers and downstream tooling (CI scripts, dashboards) triage findings faster.

## Proposed Change

Add an `IssueCategory` enum and a `category` field to `ValidationIssue` in `src/sbom_validator/models.py`:

```python
class IssueCategory(StrEnum):
    SCHEMA = "SCHEMA"        # JSON schema / XSD violation
    NTIA   = "NTIA"          # NTIA minimum element missing
    FORMAT = "FORMAT"        # Format detection / parse failure
```

```python
@dataclass(frozen=True)
class ValidationIssue:
    severity:   IssueSeverity
    category:   IssueCategory   # ← new field
    field_path: str
    message:    str
    rule:       str = ""
```

Populated by:
- `schema_validator.py` → `IssueCategory.SCHEMA`
- `ntia_checker.py`     → `IssueCategory.NTIA`
- format detection / parser errors → `IssueCategory.FORMAT`

### Text output (`--format text`)

Group issues under labelled headers, e.g.:

```
Schema Errors (2)
  [ERROR] components[3].version — Field 'version' is required …

NTIA Compliance Issues (1)
  [ERROR] components[3] — Supplier information is missing …
```

### JSON output (`--format json`)

Each issue object gains a `"category"` key:

```json
{
  "severity": "ERROR",
  "category": "SCHEMA",
  "field_path": "components[3].version",
  "message": "Field 'version' is required …",
  "rule": "CDX-SCHEMA-001"
}
```

## Acceptance Criteria

- [ ] `IssueCategory` enum added to `models.py`
- [ ] `ValidationIssue.category` field populated by all three producers
- [ ] Text output groups issues by category with labelled sections
- [ ] JSON output includes `"category"` key on every issue object
- [ ] Existing tests updated; new tests cover each category path
- [ ] `--format json` output schema stays backward-compatible (additive change)
- [ ] Quality gate passes: ruff, mypy, pytest ≥ 90% coverage

## Notes

- The `category` addition to `ValidationIssue` is a **breaking change to the frozen dataclass** — all call sites that construct `ValidationIssue` directly must be updated.
- The JSON key is additive, so existing consumers that ignore unknown keys are unaffected. Consumers that validate strict JSON schemas will need updating.
- Consider whether `IssueCategory` should be surfaced in the report writer (`report_writer.py`) HTML output as well.

--- a/src/sbom_validator/cli.py
+++ b/src/sbom_validator/cli.py
@@ -12,7 +12,7 @@ import click
 
 from sbom_validator import __version__
 from sbom_validator.logging_config import configure_logging
-from sbom_validator.models import ValidationResult, ValidationStatus
+from sbom_validator.models import IssueCategory, ValidationIssue, ValidationResult, ValidationStatus
 from sbom_validator.presentation import humanize_field_path, humanize_message
 from sbom_validator.report_writer import write_reports
 from sbom_validator.validator import validate
@@ -30,6 +30,7 @@ def _result_to_dict(result: ValidationResult) -> dict[str, Any]:
         "issues": [
             {
                 "severity": issue.severity.value,
+                "category": issue.category.value,
                 "field_path": issue.field_path,
                 "message": issue.message,
                 "rule": issue.rule,
@@ -48,11 +49,20 @@ def _exit_code(result: ValidationResult) -> int:
     return 2  # ERROR
 
 
+_CATEGORY_LABELS: dict[str, str] = {
+    IssueCategory.SCHEMA: "Schema Issues",
+    IssueCategory.NTIA: "NTIA Compliance Issues",
+    IssueCategory.FORMAT: "Format / Detection Errors",
+}
+
+_CATEGORY_ORDER: list[str] = [IssueCategory.FORMAT, IssueCategory.SCHEMA, IssueCategory.NTIA]
+
+
 def _render_text(result: ValidationResult) -> str:
     """Render a human-readable text report.
 
     Rule IDs are intentionally omitted from text output to reduce
-    implementation-detail noise for end users.
+    implementation-detail noise for end users. Issues are grouped by category.
     """
     lines: list[str] = []
     status_label = result.status.value  # "PASS", "FAIL", "ERROR"
@@ -62,11 +72,20 @@ def _render_text(result: ValidationResult) -> str:
         lines.append(f"Format:  {result.format_detected}")
     if result.issues:
         lines.append(f"Issues:  {len(result.issues)}")
+        grouped: dict[str, list[ValidationIssue]] = {}
         for issue in result.issues:
-            lines.append(
-                f"  [{issue.severity.value}] {humanize_field_path(issue.field_path)}: "
-                f"{humanize_message(issue.message)}"
-            )
+            grouped.setdefault(issue.category.value, []).append(issue)
+        for cat in _CATEGORY_ORDER:
+            cat_issues = grouped.get(cat, [])
+            if not cat_issues:
+                continue
+            label = _CATEGORY_LABELS.get(cat, cat)
+            lines.append(f"\n{label} ({len(cat_issues)})")
+            for issue in cat_issues:
+                lines.append(
+                    f"  [{issue.severity.value}] {humanize_field_path(issue.field_path)}: "
+                    f"{humanize_message(issue.message)}"
+                )
     else:
         if result.status == ValidationStatus.PASS:
             lines.append("Issues:  none")

--- a/src/sbom_validator/cli.py
+++ b/src/sbom_validator/cli.py
@@ -11,8 +11,9 @@ from typing import Any
 import click
 
 from sbom_validator import __version__
+from sbom_validator.constants import CATEGORY_LABELS, CATEGORY_ORDER
 from sbom_validator.logging_config import configure_logging
-from sbom_validator.models import IssueCategory, ValidationIssue, ValidationResult, ValidationStatus
+from sbom_validator.models import ValidationIssue, ValidationResult, ValidationStatus
 from sbom_validator.presentation import humanize_field_path, humanize_message
 from sbom_validator.report_writer import write_reports
 from sbom_validator.validator import validate
@@ -49,15 +50,6 @@ def _exit_code(result: ValidationResult) -> int:
     return 2  # ERROR
 
 
-_CATEGORY_LABELS: dict[str, str] = {
-    IssueCategory.SCHEMA: "Schema Issues",
-    IssueCategory.NTIA: "NTIA Compliance Issues",
-    IssueCategory.FORMAT: "Format / Detection Errors",
-}
-
-_CATEGORY_ORDER: list[str] = [IssueCategory.FORMAT, IssueCategory.SCHEMA, IssueCategory.NTIA]
-
-
 def _render_text(result: ValidationResult) -> str:
     """Render a human-readable text report.
 
@@ -75,11 +67,11 @@ def _render_text(result: ValidationResult) -> str:
         grouped: dict[str, list[ValidationIssue]] = {}
         for issue in result.issues:
             grouped.setdefault(issue.category.value, []).append(issue)
-        for cat in _CATEGORY_ORDER:
+        for cat in CATEGORY_ORDER:
             cat_issues = grouped.get(cat, [])
             if not cat_issues:
                 continue
-            label = _CATEGORY_LABELS.get(cat, cat)
+            label = CATEGORY_LABELS.get(cat, cat)
             lines.append(f"\n{label} ({len(cat_issues)})")
             for issue in cat_issues:
                 lines.append(

--- a/src/sbom_validator/constants.py
+++ b/src/sbom_validator/constants.py
@@ -5,6 +5,8 @@ validation rule codes, supported version strings) must be defined here and
 imported by the modules that need them.
 """
 
+from sbom_validator.models import IssueCategory
+
 # ── Format identifiers ─────────────────────────────────────────────────────
 # Returned by detect_format() and threaded through the whole pipeline.
 FORMAT_SPDX = "spdx"  # SPDX 2.3 JSON
@@ -37,3 +39,11 @@ RULE_IDENTIFIERS = "FR-07"
 RULE_RELATIONSHIPS = "FR-08"
 RULE_AUTHOR = "FR-09"
 RULE_TIMESTAMP = "FR-10"
+
+# ── Issue category display constants ───────────────────────────────────────
+CATEGORY_LABELS: dict[str, str] = {
+    IssueCategory.FORMAT: "Format / Detection Errors",
+    IssueCategory.SCHEMA: "Schema Issues",
+    IssueCategory.NTIA: "NTIA Compliance Issues",
+}
+CATEGORY_ORDER: list[str] = [IssueCategory.FORMAT, IssueCategory.SCHEMA, IssueCategory.NTIA]

--- a/src/sbom_validator/models.py
+++ b/src/sbom_validator/models.py
@@ -22,11 +22,20 @@ class IssueSeverity(StrEnum):
     INFO = "INFO"
 
 
+class IssueCategory(StrEnum):
+    """Category of a validation issue — which pipeline stage produced it."""
+
+    SCHEMA = "SCHEMA"
+    NTIA = "NTIA"
+    FORMAT = "FORMAT"
+
+
 @dataclass(frozen=True)
 class ValidationIssue:
     """A single validation finding."""
 
     severity: IssueSeverity
+    category: IssueCategory
     field_path: str
     message: str
     rule: str = ""

--- a/src/sbom_validator/ntia_checker.py
+++ b/src/sbom_validator/ntia_checker.py
@@ -14,7 +14,7 @@ from sbom_validator.constants import (
     RULE_TIMESTAMP,
     RULE_VERSION,
 )
-from sbom_validator.models import IssueSeverity, NormalizedSBOM, ValidationIssue
+from sbom_validator.models import IssueCategory, IssueSeverity, NormalizedSBOM, ValidationIssue
 
 logger = logging.getLogger(__name__)
 
@@ -46,6 +46,7 @@ def _check_supplier(sbom: NormalizedSBOM) -> list[ValidationIssue]:
             issues.append(
                 ValidationIssue(
                     severity=IssueSeverity.ERROR,
+                    category=IssueCategory.NTIA,
                     field_path=f"components[{i}].supplier",
                     message=(
                         f"Component '{component.name}' is missing a supplier name (NTIA FR-04)"
@@ -64,6 +65,7 @@ def _check_component_name(sbom: NormalizedSBOM) -> list[ValidationIssue]:
             issues.append(
                 ValidationIssue(
                     severity=IssueSeverity.ERROR,
+                    category=IssueCategory.NTIA,
                     field_path=f"components[{i}].name",
                     message=(f"Component at index {i} is missing a component name (NTIA FR-05)"),
                     rule=RULE_COMPONENT_NAME,
@@ -80,6 +82,7 @@ def _check_version(sbom: NormalizedSBOM) -> list[ValidationIssue]:
             issues.append(
                 ValidationIssue(
                     severity=IssueSeverity.ERROR,
+                    category=IssueCategory.NTIA,
                     field_path=f"components[{i}].version",
                     message=(f"Component '{component.name}' is missing a version (NTIA FR-06)"),
                     rule=RULE_VERSION,
@@ -96,6 +99,7 @@ def _check_identifiers(sbom: NormalizedSBOM) -> list[ValidationIssue]:
             issues.append(
                 ValidationIssue(
                     severity=IssueSeverity.ERROR,
+                    category=IssueCategory.NTIA,
                     field_path=f"components[{i}].identifiers",
                     message=(
                         f"Component '{component.name}' has no unique identifiers "
@@ -113,6 +117,7 @@ def _check_relationships(sbom: NormalizedSBOM) -> list[ValidationIssue]:
         return [
             ValidationIssue(
                 severity=IssueSeverity.ERROR,
+                category=IssueCategory.NTIA,
                 field_path="relationships",
                 message=("SBOM declares no dependency relationships (NTIA FR-08)"),
                 rule=RULE_RELATIONSHIPS,
@@ -127,6 +132,7 @@ def _check_author(sbom: NormalizedSBOM) -> list[ValidationIssue]:
         return [
             ValidationIssue(
                 severity=IssueSeverity.ERROR,
+                category=IssueCategory.NTIA,
                 field_path="author",
                 message=("SBOM is missing author information (NTIA FR-09)"),
                 rule=RULE_AUTHOR,
@@ -141,6 +147,7 @@ def _check_timestamp(sbom: NormalizedSBOM) -> list[ValidationIssue]:
         return [
             ValidationIssue(
                 severity=IssueSeverity.ERROR,
+                category=IssueCategory.NTIA,
                 field_path="timestamp",
                 message=("SBOM is missing a creation timestamp (NTIA FR-10)"),
                 rule=RULE_TIMESTAMP,
@@ -156,6 +163,7 @@ def _check_timestamp(sbom: NormalizedSBOM) -> list[ValidationIssue]:
         return [
             ValidationIssue(
                 severity=IssueSeverity.ERROR,
+                category=IssueCategory.NTIA,
                 field_path="timestamp",
                 message=(
                     f"SBOM timestamp '{raw_timestamp}' is not a valid ISO 8601 date-time "

--- a/src/sbom_validator/report_writer.py
+++ b/src/sbom_validator/report_writer.py
@@ -14,7 +14,7 @@ from datetime import UTC, datetime
 from pathlib import Path
 
 from sbom_validator.constants import FORMAT_CYCLONEDX, FORMAT_SPDX
-from sbom_validator.models import IssueSeverity, ValidationResult
+from sbom_validator.models import IssueCategory, IssueSeverity, ValidationIssue, ValidationResult
 from sbom_validator.presentation import (
     humanize_field_path,
     humanize_message,
@@ -42,6 +42,18 @@ _SEVERITY_ORDER: dict[str, int] = {
     IssueSeverity.WARNING: 1,
     IssueSeverity.INFO: 2,
 }
+
+# ---------------------------------------------------------------------------
+# Issue category display labels and ordering
+# ---------------------------------------------------------------------------
+
+_CATEGORY_LABELS: dict[str, str] = {
+    IssueCategory.FORMAT: "Format / Detection Errors",
+    IssueCategory.SCHEMA: "Schema Issues",
+    IssueCategory.NTIA: "NTIA Compliance Issues",
+}
+
+_CATEGORY_ORDER: list[str] = [IssueCategory.FORMAT, IssueCategory.SCHEMA, IssueCategory.NTIA]
 
 # ---------------------------------------------------------------------------
 # Status badge colours (ADR-007 §HTML Report Structure)
@@ -108,6 +120,8 @@ _HTML_TEMPLATE = string.Template("""\
     .sev-WARNING { color: #e65100; font-weight: bold; }
     .sev-INFO    { color: #1565c0; }
     .no-issues { color: #2e7d32; font-weight: bold; margin: 1rem 0; }
+    .category-section { margin-bottom: 1.5rem; }
+    .category-section h3 { font-size: 1rem; margin: 0 0 0.5rem 0; color: #424242; border-left: 4px solid #424242; padding-left: 0.5rem; }
     footer { margin-top: 2rem; font-size: 0.8rem; color: #999; border-top: 1px solid #e0e0e0; padding-top: 0.5rem; }
   </style>
 </head>
@@ -238,29 +252,43 @@ def write_reports(
     format_display = format_detected_expanded if format_detected_expanded is not None else "Unknown"
 
     if sorted_issues:
-        rows_parts: list[str] = []
+        grouped: dict[str, list[ValidationIssue]] = {}
         for issue in sorted_issues:
-            friendly_message = humanize_message(issue.message)
-            base_message, hint = split_message_and_hint(friendly_message)
-            rows_parts.append(
-                f"    <tr>"
-                f'<td class="sev-{_escape(str(issue.severity))}">{_escape(str(issue.severity))}</td>'
-                f"<td>{_escape(humanize_field_path(issue.field_path))}</td>"
-                f"<td>{_escape(base_message)}</td>"
-                f"<td>{_escape(hint or '-')}</td>"
-                f"</tr>"
+            grouped.setdefault(str(issue.category), []).append(issue)
+
+        section_parts: list[str] = []
+        for cat in _CATEGORY_ORDER:
+            cat_issues = grouped.get(cat, [])
+            if not cat_issues:
+                continue
+            label = _CATEGORY_LABELS.get(cat, cat)
+            rows_parts: list[str] = []
+            for issue in cat_issues:
+                friendly_message = humanize_message(issue.message)
+                base_message, hint = split_message_and_hint(friendly_message)
+                rows_parts.append(
+                    f"    <tr>"
+                    f'<td class="sev-{_escape(str(issue.severity))}">{_escape(str(issue.severity))}</td>'
+                    f"<td>{_escape(humanize_field_path(issue.field_path))}</td>"
+                    f"<td>{_escape(base_message)}</td>"
+                    f"<td>{_escape(hint or '-')}</td>"
+                    f"</tr>"
+                )
+            rows = "\n".join(rows_parts)
+            section_parts.append(
+                f'<div class="category-section">\n'
+                f"  <h3>{_escape(label)} ({len(cat_issues)})</h3>\n"
+                f"  <table>\n"
+                f"    <thead>\n"
+                f"      <tr><th>Severity</th><th>Field Path</th><th>Message</th><th>Hint</th></tr>\n"
+                f"    </thead>\n"
+                f"    <tbody>\n"
+                f"{rows}\n"
+                f"    </tbody>\n"
+                f"  </table>\n"
+                f"</div>"
             )
-        rows = "\n".join(rows_parts)
-        issues_section = (
-            "<table>\n"
-            "  <thead>\n"
-            "    <tr><th>Severity</th><th>Field Path</th><th>Message</th><th>Hint</th></tr>\n"
-            "  </thead>\n"
-            "  <tbody>\n"
-            f"{rows}\n"
-            "  </tbody>\n"
-            "</table>"
-        )
+        issues_section = "\n".join(section_parts)
     else:
         issues_section = '<p class="no-issues">No issues found.</p>'
 
@@ -295,6 +323,7 @@ def write_reports(
         "issues": [
             {
                 "severity": str(issue.severity),
+                "category": issue.category.value,
                 "rule": issue.rule,
                 "field_path": issue.field_path,
                 "message": issue.message,

--- a/src/sbom_validator/report_writer.py
+++ b/src/sbom_validator/report_writer.py
@@ -13,8 +13,8 @@ import string
 from datetime import UTC, datetime
 from pathlib import Path
 
-from sbom_validator.constants import FORMAT_CYCLONEDX, FORMAT_SPDX
-from sbom_validator.models import IssueCategory, IssueSeverity, ValidationIssue, ValidationResult
+from sbom_validator.constants import CATEGORY_LABELS, CATEGORY_ORDER, FORMAT_CYCLONEDX, FORMAT_SPDX
+from sbom_validator.models import IssueSeverity, ValidationIssue, ValidationResult
 from sbom_validator.presentation import (
     humanize_field_path,
     humanize_message,
@@ -42,18 +42,6 @@ _SEVERITY_ORDER: dict[str, int] = {
     IssueSeverity.WARNING: 1,
     IssueSeverity.INFO: 2,
 }
-
-# ---------------------------------------------------------------------------
-# Issue category display labels and ordering
-# ---------------------------------------------------------------------------
-
-_CATEGORY_LABELS: dict[str, str] = {
-    IssueCategory.FORMAT: "Format / Detection Errors",
-    IssueCategory.SCHEMA: "Schema Issues",
-    IssueCategory.NTIA: "NTIA Compliance Issues",
-}
-
-_CATEGORY_ORDER: list[str] = [IssueCategory.FORMAT, IssueCategory.SCHEMA, IssueCategory.NTIA]
 
 # ---------------------------------------------------------------------------
 # Status badge colours (ADR-007 §HTML Report Structure)
@@ -257,11 +245,11 @@ def write_reports(
             grouped.setdefault(str(issue.category), []).append(issue)
 
         section_parts: list[str] = []
-        for cat in _CATEGORY_ORDER:
+        for cat in CATEGORY_ORDER:
             cat_issues = grouped.get(cat, [])
             if not cat_issues:
                 continue
-            label = _CATEGORY_LABELS.get(cat, cat)
+            label = CATEGORY_LABELS.get(cat, cat)
             rows_parts: list[str] = []
             for issue in cat_issues:
                 friendly_message = humanize_message(issue.message)

--- a/src/sbom_validator/schema_validator.py
+++ b/src/sbom_validator/schema_validator.py
@@ -22,7 +22,7 @@ from sbom_validator.constants import (
     RULE_CDX_SCHEMA,
     RULE_SPDX_SCHEMA,
 )
-from sbom_validator.models import IssueSeverity, ValidationIssue
+from sbom_validator.models import IssueCategory, IssueSeverity, ValidationIssue
 
 logger = logging.getLogger(__name__)
 
@@ -90,6 +90,7 @@ def _validate_json_schema(
         issues.append(
             ValidationIssue(
                 severity=IssueSeverity.ERROR,
+                category=IssueCategory.SCHEMA,
                 field_path=field_path,
                 message=error.message,
                 rule=rule,
@@ -107,6 +108,7 @@ def _validate_cyclonedx_xml(raw_doc: str, cdx_version: str, rule: str) -> list[V
         return [
             ValidationIssue(
                 severity=IssueSeverity.ERROR,
+                category=IssueCategory.SCHEMA,
                 field_path="$",
                 message=f"Invalid XML: {exc}",
                 rule=rule,
@@ -119,6 +121,7 @@ def _validate_cyclonedx_xml(raw_doc: str, cdx_version: str, rule: str) -> list[V
         issues.append(
             ValidationIssue(
                 severity=IssueSeverity.ERROR,
+                category=IssueCategory.SCHEMA,
                 field_path=field_path,
                 message=error.reason or str(error),
                 rule=rule,

--- a/src/sbom_validator/validator.py
+++ b/src/sbom_validator/validator.py
@@ -20,6 +20,7 @@ from sbom_validator.constants import (
 from sbom_validator.exceptions import ParseError, UnsupportedFormatError
 from sbom_validator.format_detector import detect_format
 from sbom_validator.models import (
+    IssueCategory,
     IssueSeverity,
     ValidationIssue,
     ValidationResult,
@@ -58,6 +59,7 @@ def _error_issue(message: str, rule: str = "SYS-ERROR") -> ValidationIssue:
     """Build a consistent ERROR issue payload for tool/input failures."""
     return ValidationIssue(
         severity=IssueSeverity.ERROR,
+        category=IssueCategory.FORMAT,
         field_path="",
         message=message,
         rule=rule,

--- a/tests/unit/test_cli_json_output.py
+++ b/tests/unit/test_cli_json_output.py
@@ -233,12 +233,29 @@ class TestCliJsonOutputFail:
         assert "rule" in issue
 
     def test_failing_spdx_json_issue_required_keys_all_present(self, runner: CliRunner) -> None:
-        """Single assertion verifying all four required keys exist on every issue."""
+        """Single assertion verifying all required keys exist on every issue."""
         result = self._invoke_missing_supplier(runner)
         data = json.loads(result.output)
         for issue in data["issues"]:
-            for key in ("severity", "field_path", "message", "rule"):
+            for key in ("severity", "category", "field_path", "message", "rule"):
                 assert key in issue, f"Issue is missing key '{key}': {issue}"
+
+    def test_failing_spdx_json_issue_has_category_key(self, runner: CliRunner) -> None:
+        result = self._invoke_missing_supplier(runner)
+        data = json.loads(result.output)
+        assert "category" in data["issues"][0]
+
+    def test_failing_spdx_json_issue_category_is_valid_value(self, runner: CliRunner) -> None:
+        result = self._invoke_missing_supplier(runner)
+        data = json.loads(result.output)
+        valid_categories = {"SCHEMA", "NTIA", "FORMAT"}
+        for issue in data["issues"]:
+            assert issue["category"] in valid_categories
+
+    def test_failing_spdx_json_ntia_issue_category_is_ntia(self, runner: CliRunner) -> None:
+        result = self._invoke_missing_supplier(runner)
+        data = json.loads(result.output)
+        assert all(issue["category"] == "NTIA" for issue in data["issues"])
 
     def test_failing_spdx_json_issue_severity_is_valid_value(self, runner: CliRunner) -> None:
         result = self._invoke_missing_supplier(runner)

--- a/tests/unit/test_cli_text_output.py
+++ b/tests/unit/test_cli_text_output.py
@@ -2,6 +2,7 @@
 
 from __future__ import annotations
 
+import re
 from pathlib import Path
 
 from click.testing import CliRunner
@@ -214,3 +215,28 @@ class TestCliValidateErrorCases:
         f.write_text("{}")
         result = runner.invoke(main, ["validate", str(f)])
         assert result.exit_code == 2
+
+
+# ===========================================================================
+# TestCliTextCategoryGrouping
+# ===========================================================================
+
+
+class TestCliTextCategoryGrouping:
+    """Text output groups issues under labelled category headers."""
+
+    def test_ntia_failure_shows_ntia_compliance_issues_header(self, runner: CliRunner) -> None:
+        result = runner.invoke(
+            main, ["validate", str(SPDX_FIXTURES / "missing-supplier.spdx.json")]
+        )
+        assert "NTIA Compliance Issues" in result.output
+
+    def test_schema_failure_shows_schema_issues_header(self, runner: CliRunner) -> None:
+        result = runner.invoke(main, ["validate", str(SPDX_FIXTURES / "invalid-schema.spdx.json")])
+        assert "Schema Issues" in result.output
+
+    def test_category_header_includes_count(self, runner: CliRunner) -> None:
+        result = runner.invoke(
+            main, ["validate", str(SPDX_FIXTURES / "missing-supplier.spdx.json")]
+        )
+        assert re.search(r"NTIA Compliance Issues \(\d+\)", result.output)

--- a/tests/unit/test_models.py
+++ b/tests/unit/test_models.py
@@ -12,6 +12,7 @@ from dataclasses import FrozenInstanceError
 import pytest
 
 from sbom_validator.models import (
+    IssueCategory,
     IssueSeverity,
     NormalizedComponent,
     NormalizedRelationship,
@@ -57,6 +58,25 @@ class TestIssueSeverity:
 
 
 # ---------------------------------------------------------------------------
+# IssueCategory
+# ---------------------------------------------------------------------------
+
+
+class TestIssueCategory:
+    def test_schema_value_equals_string(self):
+        assert IssueCategory.SCHEMA.value == "SCHEMA"
+
+    def test_ntia_value_equals_string(self):
+        assert IssueCategory.NTIA.value == "NTIA"
+
+    def test_format_value_equals_string(self):
+        assert IssueCategory.FORMAT.value == "FORMAT"
+
+    def test_has_exactly_three_members(self):
+        assert len(IssueCategory) == 3
+
+
+# ---------------------------------------------------------------------------
 # ValidationIssue
 # ---------------------------------------------------------------------------
 
@@ -65,18 +85,21 @@ class TestValidationIssue:
     def test_instantiation_with_all_fields(self):
         issue = ValidationIssue(
             severity=IssueSeverity.ERROR,
+            category=IssueCategory.SCHEMA,
             field_path="packages[0].name",
             message="Name is missing",
             rule="PKG-001",
         )
         assert issue.severity == IssueSeverity.ERROR
+        assert issue.category == IssueCategory.SCHEMA
         assert issue.field_path == "packages[0].name"
         assert issue.message == "Name is missing"
         assert issue.rule == "PKG-001"
 
-    def test_instantiation_with_required_fields_only_rule_defaults_to_empty_string(self):
+    def test_rule_defaults_to_empty_string(self):
         issue = ValidationIssue(
             severity=IssueSeverity.WARNING,
+            category=IssueCategory.NTIA,
             field_path="metadata.author",
             message="Author not provided",
         )
@@ -85,6 +108,7 @@ class TestValidationIssue:
     def test_immutability_raises_frozen_instance_error(self):
         issue = ValidationIssue(
             severity=IssueSeverity.INFO,
+            category=IssueCategory.FORMAT,
             field_path="root",
             message="Info note",
         )
@@ -94,26 +118,45 @@ class TestValidationIssue:
     def test_equality_same_fields_are_equal(self):
         issue_a = ValidationIssue(
             severity=IssueSeverity.ERROR,
+            category=IssueCategory.SCHEMA,
             field_path="foo.bar",
             message="Something wrong",
             rule="R-01",
         )
         issue_b = ValidationIssue(
             severity=IssueSeverity.ERROR,
+            category=IssueCategory.SCHEMA,
             field_path="foo.bar",
             message="Something wrong",
             rule="R-01",
         )
         assert issue_a == issue_b
 
-    def test_equality_different_fields_are_not_equal(self):
+    def test_equality_different_category_not_equal(self):
         issue_a = ValidationIssue(
             severity=IssueSeverity.ERROR,
+            category=IssueCategory.SCHEMA,
+            field_path="foo",
+            message="msg",
+        )
+        issue_b = ValidationIssue(
+            severity=IssueSeverity.ERROR,
+            category=IssueCategory.NTIA,
+            field_path="foo",
+            message="msg",
+        )
+        assert issue_a != issue_b
+
+    def test_equality_different_severity_not_equal(self):
+        issue_a = ValidationIssue(
+            severity=IssueSeverity.ERROR,
+            category=IssueCategory.NTIA,
             field_path="foo",
             message="msg",
         )
         issue_b = ValidationIssue(
             severity=IssueSeverity.WARNING,
+            category=IssueCategory.NTIA,
             field_path="foo",
             message="msg",
         )
@@ -240,6 +283,7 @@ class TestValidationResult:
     def test_instantiation_with_all_fields(self):
         issue = ValidationIssue(
             severity=IssueSeverity.ERROR,
+            category=IssueCategory.SCHEMA,
             field_path="root",
             message="Missing required field",
         )

--- a/tests/unit/test_report_writer.py
+++ b/tests/unit/test_report_writer.py
@@ -20,7 +20,13 @@ import re
 from pathlib import Path
 from unittest.mock import patch
 
-from sbom_validator.models import IssueSeverity, ValidationIssue, ValidationResult, ValidationStatus
+from sbom_validator.models import (
+    IssueCategory,
+    IssueSeverity,
+    ValidationIssue,
+    ValidationResult,
+    ValidationStatus,
+)
 from sbom_validator.report_writer import write_reports  # will cause ImportError until implemented
 
 # ---------------------------------------------------------------------------
@@ -29,18 +35,21 @@ from sbom_validator.report_writer import write_reports  # will cause ImportError
 
 _ISSUE_ERROR = ValidationIssue(
     severity=IssueSeverity.ERROR,
+    category=IssueCategory.SCHEMA,
     field_path="packages.0.name",
     message="Field 'name' is required",
     rule="FR-02",
 )
 _ISSUE_WARNING = ValidationIssue(
     severity=IssueSeverity.WARNING,
+    category=IssueCategory.NTIA,
     field_path="packages.0.supplier",
     message="Supplier is missing",
     rule="FR-06",
 )
 _ISSUE_INFO = ValidationIssue(
     severity=IssueSeverity.INFO,
+    category=IssueCategory.NTIA,
     field_path="creationInfo",
     message="Optional field absent",
     rule="FR-09",
@@ -416,6 +425,7 @@ class TestHtmlReportContent:
     def test_html_humanizes_xml_field_paths_and_messages(self, tmp_path: Path) -> None:
         xml_issue = ValidationIssue(
             severity=IssueSeverity.ERROR,
+            category=IssueCategory.SCHEMA,
             field_path=(
                 "/{http://cyclonedx.org/schema/bom/1.6}bom/"
                 "{http://cyclonedx.org/schema/bom/1.6}components/"
@@ -441,6 +451,7 @@ class TestHtmlReportContent:
     def test_html_hides_ntia_rule_ids_and_adds_supplier_hint(self, tmp_path: Path) -> None:
         ntia_issue = ValidationIssue(
             severity=IssueSeverity.ERROR,
+            category=IssueCategory.NTIA,
             field_path="components[0].supplier",
             message="Component 'requests' is missing a supplier name (NTIA FR-04)",
             rule="FR-04",
@@ -463,6 +474,7 @@ class TestHtmlReportContent:
     def test_html_places_message_and_hint_in_separate_columns(self, tmp_path: Path) -> None:
         xml_issue = ValidationIssue(
             severity=IssueSeverity.ERROR,
+            category=IssueCategory.SCHEMA,
             field_path="packages.0",
             message="'SPDXID' is a required property",
             rule="FR-02",
@@ -519,6 +531,108 @@ class TestReturnValue:
         result = _pass_result()
         _, json_path = write_reports(result, tmp_path)
         assert json_path.exists()
+
+
+# ===========================================================================
+# TestCategoryInReports
+# ===========================================================================
+
+
+class TestCategoryInReports:
+    """category field appears in JSON report; HTML groups issues by category."""
+
+    def _load_json(self, result: ValidationResult, tmp_path: Path) -> dict:
+        _, json_path = write_reports(result, tmp_path)
+        return json.loads(json_path.read_text(encoding="utf-8"))
+
+    def _load_html(self, result: ValidationResult, tmp_path: Path) -> str:
+        html_path, _ = write_reports(result, tmp_path)
+        return html_path.read_text(encoding="utf-8")
+
+    def test_json_issue_has_category_key(self, tmp_path: Path) -> None:
+        data = self._load_json(_fail_result(), tmp_path)
+        assert "category" in data["issues"][0]
+
+    def test_json_issue_category_schema_value(self, tmp_path: Path) -> None:
+        data = self._load_json(_fail_result(), tmp_path)
+        schema_issues = [i for i in data["issues"] if i["category"] == "SCHEMA"]
+        assert len(schema_issues) > 0
+
+    def test_json_issue_category_ntia_value(self, tmp_path: Path) -> None:
+        result = ValidationResult(
+            status=ValidationStatus.FAIL,
+            file_path="/tmp/bom.json",
+            format_detected="spdx",
+            issues=(_ISSUE_WARNING,),
+        )
+        data = self._load_json(result, tmp_path)
+        assert data["issues"][0]["category"] == "NTIA"
+
+    def test_json_issue_category_format_value(self, tmp_path: Path) -> None:
+        format_issue = ValidationIssue(
+            severity=IssueSeverity.ERROR,
+            category=IssueCategory.FORMAT,
+            field_path="",
+            message="Unsupported format",
+            rule="FR-01",
+        )
+        result = ValidationResult(
+            status=ValidationStatus.ERROR,
+            file_path="/tmp/bom.json",
+            format_detected=None,
+            issues=(format_issue,),
+        )
+        data = self._load_json(result, tmp_path)
+        assert data["issues"][0]["category"] == "FORMAT"
+
+    def test_html_contains_schema_issues_section_header(self, tmp_path: Path) -> None:
+        html = self._load_html(_fail_result(), tmp_path)
+        assert "Schema Issues" in html
+
+    def test_html_contains_ntia_issues_section_header(self, tmp_path: Path) -> None:
+        result = ValidationResult(
+            status=ValidationStatus.FAIL,
+            file_path="/tmp/bom.json",
+            format_detected="spdx",
+            issues=(_ISSUE_WARNING,),
+        )
+        html = self._load_html(result, tmp_path)
+        assert "NTIA Compliance Issues" in html
+
+    def test_html_contains_format_errors_section_header(self, tmp_path: Path) -> None:
+        format_issue = ValidationIssue(
+            severity=IssueSeverity.ERROR,
+            category=IssueCategory.FORMAT,
+            field_path="",
+            message="Unsupported format",
+            rule="FR-01",
+        )
+        result = ValidationResult(
+            status=ValidationStatus.ERROR,
+            file_path="/tmp/bom.json",
+            format_detected=None,
+            issues=(format_issue,),
+        )
+        html = self._load_html(result, tmp_path)
+        assert "Format / Detection Errors" in html
+
+    def test_html_schema_section_appears_before_ntia_section(self, tmp_path: Path) -> None:
+        result = ValidationResult(
+            status=ValidationStatus.FAIL,
+            file_path="/tmp/bom.json",
+            format_detected="spdx",
+            issues=(_ISSUE_WARNING, _ISSUE_ERROR),
+        )
+        html = self._load_html(result, tmp_path)
+        schema_pos = html.find("Schema Issues")
+        ntia_pos = html.find("NTIA Compliance Issues")
+        assert schema_pos != -1
+        assert ntia_pos != -1
+        assert schema_pos < ntia_pos
+
+    def test_html_category_section_shows_count(self, tmp_path: Path) -> None:
+        html = self._load_html(_fail_result(), tmp_path)
+        assert "(1)" in html
 
 
 # ===========================================================================

